CRM/Contribute/BAO/Contribution.php

```diff
@@ -5457,4 +5457,37 @@ public static function createProportionalFinancialEntries($entityParams, $lineIt
     }
   }
 
+  /**
+   * Calculate amounts when Financial type for contribution is changed.
+   *
+   * @param array $params
+   *   contribution params
+   * @param float $trxnAmount
+   *   financial trxn total amount
+   * @param float $totalAmount
+   *   Contribution total amount
+   *
+   * @return array
+   */
+  public static function calculateFTChangeAmount(&$params, $trxnAmount, &$totalAmount) {
+    $changeFTAmount = $trxnAmount;
+    $ignoreChangeAmount = FALSE;
+    $taxAmounts = array(
+      'new_tax_amount' => NULL,
+    );
+    if (isset($params['prevContribution']->total_amount) || isset($params['tax_amount'])) {
+      $taxAmount = CRM_Utils_Array::value('tax_amount', $params, 0);
+      $changesinTaxAmount = $totalAmount - $params['prevContribution']->total_amount + $params['prevContribution']->tax_amount - $taxAmount;
```

Comment:
Thanks, @MiyaNoctem ! @pradpnayak please review all of the logic here. It seems we need to do the calculations if a previous contribution exists, rather than if its total_amount is non-empty. Also, we need to do it regardless of whether there is tax amount on current trxn.